edxval/pacts/views.py

```diff
@@ -0,0 +1,45 @@
+"""
+Provider state views needed by pact to setup Provider state for pact verification.
+"""
+import json
+
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+
+from edxval.pacts.utils import (
+    clear_database,
+    setup_successful_video_details_state,
+    setup_successful_video_image_state,
+    setup_successful_video_status_state,
+    setup_successful_video_transcripts_state,
+    setup_unsuccessful_video_details_state,
+    setup_unsuccessful_video_image_state,
+    setup_unsuccessful_video_status_state,
+    setup_unsuccessful_video_transcripts_state,
+)
+
+
+@csrf_exempt
+@require_POST
+def provider_state(request):
+    """
+    Provider state setup view needed by pact verifier.
+    """
+    state_setup_mapping = {
+        'A valid video_id video exists': setup_successful_video_status_state,
+        'A valid video with no details exists': setup_successful_video_details_state,
+        'A valid video with no image information exists': setup_successful_video_image_state,
+        'A valid video with no video transcript information exists': setup_successful_video_transcripts_state,
+        'A valid video and video transcript information exists': setup_unsuccessful_video_transcripts_state,
+        'No video and video details exist': setup_unsuccessful_video_details_state,
+        'No valid video with video images details exist': setup_unsuccessful_video_image_state,
+        'A video against given video id does not exist': setup_unsuccessful_video_status_state,
```

Comment:
So, these are the only states that we need to test on the consumer side, right? So whenever there's a new state in provider_state, we might have to write a new test on consumer. Let me know if I was wrong anywhere.

Reply:
you got it reversed. Provider state identifies what condition the provider should be in before starting the interaction(and to make sure the interactions are independent). If a new interaction is introduced to the consumer, its appropriate state may or may not be added here.
https://docs.pact.io/getting_started/provider_states

Reply:
Ohh makes sense.